### DISCUSSION TITLE
Migration of database via Hasura

### DISCUSSION THREAD

**Discussion by M-Fontainebleau** [hasura/graphql-engine#8517] [Question] [answered]
I'm using Hasura to connect to a AWS database in my develop project. Now I want to migrate the DB in develop environment via Hasura to my staging  environment and without any operations in AWS.
is that possible with Hasura migration? I've looked up Hasura DOC and haven't found any useful information.

**Comment by seanonthenet** [comment, marked as answer]
Hi there. No you need to move existing actual data in your DB using another tool. Hasura migrations deals mainly with the SQL structure of your database not the actual data.

**Reply by seanonthenet** [reply]
If that is what you meant. If you want to move the structure then Hasura migrations cli works great.

**Reply by M-Fontainebleau** [reply]
Thank you for your answer.
Does SQL structure you said mean how I built my DB, like how many schemas are there, and within every schema how many tables are there and so on.
What I wanna do now is copy a exist DB to another empty DB(just created and without any operations)via Hasura migration. According to your answer that I can only copy the structure but not actual data to the empty DB right?

**Reply by seanonthenet** [reply]
Yeah so use the Hasura CLI migrations to move your schema. Check this out. 
https://hasura.io/docs/latest/graphql/core/migrations/index/
You can copy a little extra initial data to your database with the CLI 'seed' command too.

**Reply by M-Fontainebleau** [reply]
yes I've seen that doc and gonna try it. Thank you for your reply again.

**Reply by BenoitRanque** [reply]
You've probably resolved this by now, but in case you had not, or for any who finds this:

I'm assuming here you have a database structure you are happy with, and wish to migrate it to another environment, but do not currently have migrations for it. To do this you need to create a migration based on your current database structure.

This is covered in our doc on [setting up migrations](https://hasura.io/docs/latest/graphql/core/migrations/migrations-setup/#migrations-initialize). You can execute `hasura migrate create init --from-server` to create your init migration.
